app/models/manageiq/providers/vmware/infra_manager/ems_ref_obj_mixin.rb

```diff
@@ -0,0 +1,11 @@
+module ManageIQ::Providers::Vmware::InfraManager::EmsRefObjMixin
+  extend ActiveSupport::Concern
+
+  def ems_ref_obj
+    @ems_ref_obj = VimString.new(ems_ref, ems_ref_type, :ManagedObjectReference) if ems_ref && ems_ref != @ems_ref_obj.to_s
```

Comment:
@Fryguy how about this?  Instead of just clearing the cache on `ems_ref_obj=` this will also handle the case where `ems_ref` changed without a call to `ems_ref_obj=`